nav2_behaviors/include/nav2_behaviors/plugins/drive_on_heading.hpp

```diff
@@ -133,7 +134,35 @@ class DriveOnHeading : public TimedBehavior<ActionT>
     cmd_vel->header.frame_id = this->robot_base_frame_;
     cmd_vel->twist.linear.y = 0.0;
     cmd_vel->twist.angular.z = 0.0;
-    cmd_vel->twist.linear.x = command_speed_;
+    if (acceleration_limit_ <= 0.0 && deceleration_limit_ <= 0.0) {
```

Comment:
`||`? I assume we have to set both if we want to use either. That would let you remove the 2x `if` in the other branch as well. ALso we can simply if the `forward &&` to just `forward` and `!forward`